internal/kibana/status.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package kibana
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+const SNAPSHOT_SUFFIX = "-SNAPSHOT"
+
+type statusType struct {
+	Version struct {
+		Number        string `json:"number"`
+		BuildSnapshot bool   `json:"build_snapshot"`
+	} `json:"version"`
+}
+
+// Version method returns the version of Kibana (Elastic stack)
+func (c *Client) Version() (string, bool, error) {
+	statusCode, respBody, err := c.get(StatusAPI)
+	if err != nil {
+		return "", false, errors.Wrapf(err, "could not reach status endpoint")
+	}
+
+	if statusCode != http.StatusOK {
+		return "", false, fmt.Errorf("could not get status data; API status code = %d; response body = %s", statusCode, respBody)
+	}
+
+	var status statusType
+	err = json.Unmarshal(respBody, &status)
+	if err != nil {
+		return "", false, errors.Wrapf(err, "unmarshalling response failed (body: \n%s)", respBody)
+	}
+
+	stackVersion := status.Version.Number
+	if status.Version.BuildSnapshot {
+		stackVersion = fmt.Sprintf("%s%s", stackVersion, SNAPSHOT_SUFFIX)
+	}
+
```

Comment:
This API returns the version without `-SNAPSHOT`. https://github.com/elastic/kibana/blob/main/packages/core/status/core-status-server-internal/src/routes/status.ts#L120-L123

That's why it is needed to be added here.

Example of output (Elastic stack 7.17.0-SNAPSHOT):
```
 $ curl -u elastic:changeme -H 'Content-Type: application/json' -H 'kbn-xsrf: true' -k -s https://localhost:5601/api/status | jq -r .version
{
  "number": "7.17.0",
  "build_hash": "0ab325e4b36168a264907fbc84f6913519671acf",
  "build_number": 46537,
  "build_snapshot": true
}
```